naive_warp.py

```diff
@@ -453,23 +462,46 @@ def warp_callback(
             #     f"frontier {len(frontier)} accepted {len(acceptable)}; {len(warp_refs)} for warp"
             # )
 
-        return tuple((cmd, sum(q[1:], ())) for q in acceptable)
+        return tuple(
+            ("L", q[-1]) if _curve_is_flat(q, self._flatness) else (cmd, sum(q[1:], ()))
+            for q in acceptable
+        )
 
 
-def _quad_path_warp(view_box, warp, precision):
-    return PathWarp(view_box, warp, "Q", _warp_quad, splitQuadraticAtT, precision)
+def _quad_path_warp(view_box, warp, precision, flatness):
+    return PathWarp(
+        view_box, warp, "Q", _warp_quad, splitQuadraticAtT, precision, flatness
+    )
 
 
-def _cubic_path_warp(view_box, warp, precision):
-    return PathWarp(view_box, warp, "C", _warp_cubic, splitCubicAtT, precision)
+def _cubic_path_warp(view_box, warp, precision, flatness):
+    return PathWarp(
+        view_box, warp, "C", _warp_cubic, splitCubicAtT, precision, flatness
+    )
+
+
+def _curve_is_flat(curve, flatness=1.0):
+    # Returns True if the bezier curve is equivalent to a line.
+    # A 'flat' bezier curve is one such that the sum of the distances between
+    # consecutive control points equals the distance from start to end points.
```

Comment:
Maybe I'm just slow but I had to stop and visualize to see how this works. Consider noting this works because if a control point isn't on the line from start to end then the length will exceed the direct path start => end.